app/src/main/java/com/example/concordia_campus_guide/Adapters/RoutesAdapter.java

```diff
@@ -0,0 +1,208 @@
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import com.example.concordia_campus_guide.ClassConstants;
+import com.example.concordia_campus_guide.Models.Routes.Bus;
+import com.example.concordia_campus_guide.Models.Routes.Car;
+import com.example.concordia_campus_guide.Models.Direction;
+import com.example.concordia_campus_guide.Models.Routes.Route;
+import com.example.concordia_campus_guide.Models.Routes.Subway;
+import com.example.concordia_campus_guide.Models.Routes.Train;
+import com.example.concordia_campus_guide.Models.Routes.TransportType;
+import com.example.concordia_campus_guide.Models.Routes.Walk;
+import com.example.concordia_campus_guide.R;
+import com.google.android.flexbox.FlexboxLayout;
+
+import java.util.List;
+import androidx.annotation.NonNull;
+
+import org.w3c.dom.Text;
+
+public class RoutesAdapter extends ArrayAdapter<Route> {
+
+    List<Route> routes;
+    Context context;
+
+    // UI elements
+    ImageView mainTransportType;
+    FlexboxLayout details;
+    TextView duration;
+    LinearLayout bottom;
+    TextView arrivalAndDepartureTime;
+    LinearLayout routeOverviewLayout;
+    LinearLayout main;
+
+    public RoutesAdapter(@NonNull Context context, int textViewResourceId, @NonNull List<Route> routes) {
+        super(context, textViewResourceId, routes);
+        this.routes = routes;
+        this.context = context;
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        convertView = LayoutInflater.from(this.getContext()).inflate(R.layout.list_routes, parent, false);
+
+        routeOverviewLayout = convertView.findViewById(R.id.routeOverviewLayout);
+        main = convertView.findViewById(R.id.main);
+        mainTransportType = convertView.findViewById(R.id.mainTransportType);
+        details = convertView.findViewById(R.id.details);
+        duration = convertView.findViewById(R.id.duration);
+        bottom = convertView.findViewById(R.id.bottom);
+        arrivalAndDepartureTime = convertView.findViewById(R.id.arrivalAndDepartureTime);
+
+        Route route = getItem(position);
+
+        if (route!= null) { setUIElements(route); }
+        else { displayInfoMsg(); }
+
+        return convertView;
+    }
+    @Override
+    public Route getItem(int i){
+        return routes.get(i);
+    }
+    @Override
+    public int getCount(){
+        return routes.size();
+    }
+
+    // Helper methods for setting UI elements in the Routes Overview section of the Routes Activity page
+    private void setUIElements(Route route) {
+        if (route.getMainTransportType().equals(ClassConstants.WALKING)) { setUIWalk(route); }
+
+        else if (route.getMainTransportType().equals(ClassConstants.DRIVING)) { setUIDriving(route); }
+
+        else { // TRANSIT
+            setUITransit(route);
+
+            // Go through each steps in the route
+            for(int i = 0; i < route.getSteps().size(); i++) {
```

Comment:
looks like a switch case is better no? also the code inside the transit case could be put in its own helper method and will be cleaner!

Reply:
but step is a TransportType object. I am checking the instance of "step". Switch cases work with primitive values so I thought i'd just go with if/else